packages/vscode-extension/src/debug/depsChecker/funcToolChecker.ts

```diff
@@ -85,6 +85,15 @@ export class FuncToolChecker implements IDepsChecker {
         portableFuncVersion !== null && supportedVersions.includes(portableFuncVersion);
       // to avoid "func -v" and "func new" work well, but "func start" fail.
       hasSentinel = await fs.pathExists(FuncToolChecker.getSentinelPath());
+
+      // delete func.ps1 when checking portable function
+      for (const funcFolder in this.getPortableFuncBinFolders()) {
```

Comment:
1. Extract this to a function?
2. Delete files only when `isVersionSupported && hasSentinel` (i.e. isPortableFuncInstalled() == true) to only delete things in a known correct state?

Reply:
good point, fixed.